### ISSUE TITLE
Only kickout block / chunk producers if they are lower than threshold compared to the median

### ISSUE THREAD

**Issue by SkidanovAlex** [near/nearcore#2130]
Today we kick out block / chunk producers at the end of the epoch if the number of the blocks / chunk produced is below a certain threshold.

Since we want to set a very aggressive target block production time, some blocks are expected to get slipped. Instead we should kick out any block / chunk producer who produced below the same threshold, but compared to how many blocks / chunks were produced by a median block producer.

In other words, instead of:

```
 kickout = blocks_proudced < total_slots_allocated * threshold && (same for chunks)
 ```

we want

```
 let median = blocks_produced_by_median_bp;
 kickout = blocks_proudced < block_produced_by_median_bp * threshold && (same for chunks)
 ```

**Comment by ilblackdragon** [comment]
@jimmy3dita Apparently we decided not to do with this for MainNet and will review later.

**Comment by jimmy3dita** [comment]
Please mention me on the relevant specs, so I can quickly update docs, specs and other material
